account_financial_risk/models/account_invoice.py

```diff
@@ -1,27 +1,52 @@
 # Copyright 2016-2018 Tecnativa - Carlos Dauden
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import _, api, models
+from odoo import _, api, fields, models
 
 
 class AccountInvoice(models.Model):
     _inherit = 'account.invoice'
 
+    company_currency_id = fields.Many2one(store=True)
```

Comment:
I'm worried about the performance of this when upgrading/installing. Please include a migration script and a pre-init hook that pre-compute this field.

Reply:
I put the migration script in v3.0.0, assuming you will do a "ocabot merge major"